aiChatterBox/aiChatterBox/MainWindow.xaml.cs

```diff
@@ -0,0 +1,284 @@
+            addMessageToListView("ME: " + inputPrompt, userMessageColour, HorizontalAlignment.Right);
+            textBox_PromptInput.Text = "";
+
+            // Start Loading Animation
+            if (currentCancellationTokenSource != null)
+            {
+                currentCancellationTokenSource.Cancel();
+            }
+            currentCancellationTokenSource = new CancellationTokenSource();
+            var token = currentCancellationTokenSource.Token;
+            showLoadingAnimation(token);
+            string aiOutput = "Error: An error has occured";
+            //Get Response From Ollama, or Show Error
+            try
+            {
+                aiOutput = await promptOllamaAsync(inputPrompt);
+                chats[currentChatIndex].Add($"AI: {aiOutput}");
+                addMessageToListView("AI: " + aiOutput, llamaMessageColour, HorizontalAlignment.Left);
+            }
+            catch (Exception ex)
+            {
+                chats[currentChatIndex].Add($"Error: {ex.Message}");
+                addMessageToListView($"Error: {ex.Message}", errorMessageColour, HorizontalAlignment.Left);
+            }
+            finally
+            {
+                //Reset Values so Loading Animation can be Reused
+                currentCancellationTokenSource.Cancel();
+                currentCancellationTokenSource = null;
+
+                //Save chats to file after each interaction
+                SaveChatsToFile();
+            }
+        }
+
+        //Button Click to Submit Prompt
+        private async void button_PromptSubmit_Click(object sender, RoutedEventArgs e)
+        {
+            submitPrompt();
+        }
+
+        // Save chats to file
+        private void SaveChatsToFile()
+        {
+            string chatsJson = JsonConvert.SerializeObject(chats);
+            File.WriteAllText(chatsFilePath, chatsJson);
+        }
+
+        // Load chats from file
+        private void LoadChatsFromFile()
+        {
+            if (File.Exists(chatsFilePath))
+            {
+                string chatsJson = File.ReadAllText(chatsFilePath);
+                chats = JsonConvert.DeserializeObject<List<List<string>>>(chatsJson);
+                for (int i = 0; i < chats.Count; i++)
+                {
+                    listView_PastChats.Items.Add($"             Chat {i + 1}");
+                }
+            }
+        }
+
+        //Calling Ollama
+        private async Task<string> promptOllamaAsync(string userPrompt)
+        {
+            //Operation to Attach to Ollama host
+            string path = "/api/generate";
+            string apiUrl = localhost + path;
+
+            //Structure Request
+            var requestBody = new
+            {
+                model = "llama3",
+                prompt = userPrompt,
+                stream = false
+            };
+
+            //Format and Send Request
+            var json = JsonConvert.SerializeObject(requestBody);
```

Comment:
Consider using System.Text instead of: using Newtonsoft.Json; as its a good idea to limit the amount of dependencies where possible. .Net has matured their built in support for JSON which covers 90% of operations.